accelerator/core/core.c

```diff
@@ -506,16 +512,20 @@ status_t ta_get_bundles_by_addr(const iota_client_service_t* const service, tryt
       bundle_transactions_new(&bundle);
       flex_trits_to_trytes(bundle_hash, NUM_TRYTES_BUNDLE, transaction_bundle(curr_tx), NUM_TRITS_BUNDLE,
                            NUM_TRITS_BUNDLE);
-      ret = ta_get_bundle(service, bundle_hash, bundle);
+      ret = ta_get_bundle(&core->iota_service, bundle_hash, bundle);
       if (ret != SC_OK) {
         ta_log_error("%d\n", ret);
         goto done;
       }
-
-      bundle_array_add(bundle_array, bundle);
-      hash243_set_add(&bundle_hash_set, transaction_bundle(curr_tx));
-      free(bundle);
     }
+    ret = ta_get_bundle(&core->iota_service, bundle_hash, bundle);
```

Comment:
Why do we need exact the same code as line 515

Reply:
It is duplicated. I will remove it.